### PR TITLE
manual nullptr removals

### PR DESCRIPTION
**Pull request by neheb**
unique_ptr has a bool overload. It's also easier to reason about.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/exif.cpp [hunk 1]
```diff
@@ -281,72 +281,39 @@ namespace Exiv2 {
...
-    {
-        return key_.get() == nullptr ? "" : key_->tagName();
-    }
+    std::string Exifdatum::tagName() const { return !key_ ? "" : key_->tagName(); }
 
-    std::string Exifdatum::tagLabel() const
-    {
-        return key_.get() == nullptr ? "" : key_->tagLabel();
-    }
+    std::string Exifdatum::tagLabel() const { return !key_ ? "" : key_->tagLabel(); }
```
Comment (neheb) [inline src/exif.cpp:301]: Conditions like these should probably be flipped. eg:

```
return key_ ? key_->tagLabel() : "";
```

Comment (neheb) [reply src/exif.cpp]: Done.

File: src/exif.cpp [hunk 2]
```diff
@@ -278,72 +278,36 @@ namespace Exiv2 {
...
             value_ = Value::create(type);
         }
         return value_->read(value);
     }
 
-    int Exifdatum::setDataArea(const byte* buf, size_t len)
-    {
-        return value_.get() == nullptr ? -1 : value_->setDataArea(buf, len);
-    }
+    int Exifdatum::setDataArea(const byte* buf, size_t len) { return value_ ? value_->setDataArea(buf, len) : -1; }
```
Comment (piponazo) [inline src/exif.cpp:288]: I think this formatting style is not matching with the current clang-format settings we have in the project.

Anyways, it is not a big deal. We should probably pass clang-format over all the project soon and try to enforce keeping the format in the CI.

**Review by piponazo — APPROVED** [review]
LGTM! Thanks!

File: src/exif.cpp (hunk 2 above)
Comment (neheb) [reply src/exif.cpp:288]: That would be ideal, yes.

**Pull request merged**